torch/fx/graph_module.py

```diff
@@ -277,6 +265,186 @@ def _gen_model_repr(module_name: str, module: torch.nn.Module) -> Optional[str]:
             warnings.warn("Was not able to save the following children modules as reprs -"
                           f"saved as pickled files instead: {blobified_modules}")
 
+    def has_submodule(self, target: str) -> bool:
+        """
+        Returns whether or not this GraphModule contains the submodule
+        given by ``target``.
+
+        For example, let's say you have an ``nn.Module`` ``A`` that
+        looks like this:
+
+        .. code-block::text
+
+            A(
+                (net_b): Module(
+                    (net_c): Module(
+                        (conv): Conv2d(16, 33, kernel_size=(3, 3), stride=(2, 2))
+                    )
+                    (linear): Linear(in_features=100, out_features=200, bias=True)
+                )
+            )
+
+        (The diagram shows an ``nn.Module`` ``A``. ``A`` has a nested 
+        submodule ``net_b``, which itself has two submodules ``net_c`` 
+        and ``linear``. ``net_c`` then has a submodule ``conv``.)
+
+        To check whether or not we have the ``linear`` submodule, we
+        would call ``has_submodule("net_b.linear")``. To check whether
+        we have the ``conv`` submodule, we would call
+        ``has_submodule("net_b.net_c.conv")``.
+
+        Args:
+            target: The fully-qualified string name of the submodule
+                to look for. (See above example for how to specify a
+                fully-qualified string.)
+
+        Returns:
+            bool: Whether or not the target string referenced an
+                existing submodule. Returns False if the target string
+                resolves to something that is not an ``nn.Module``.
+        """
+        atoms: List[str] = target.split(".")
+
+        for item in atoms:
+
+            if not hasattr(self, item):
+                return False
+
+            self = getattr(self, item)
+
+            if not isinstance(self, torch.nn.Module):
+                return False
+
+        return True
```

Comment:
This could also go on `nn.Module`; nothing here is specific to `GraphModule`.

I think another way to do this with the current Module API is `target in dict(mod.named_modules())` but that's O(number of children modules) and this is probably faster. Might be a good addition to `nn.Module`